java/cdk/src/test/java/sleeper/cdk/UtilsTestHelper.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2022-2023 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package sleeper.cdk;
+
+import sleeper.configuration.properties.InstanceProperties;
+import sleeper.configuration.properties.table.TableProperties;
+
+import java.nio.file.Path;
+import java.util.Map;
+import java.util.UUID;
+import java.util.function.Function;
+
+import static sleeper.configuration.properties.SystemDefinedInstanceProperty.VERSION;
+import static sleeper.configuration.properties.UserDefinedInstanceProperty.ACCOUNT;
+import static sleeper.configuration.properties.UserDefinedInstanceProperty.ID;
+import static sleeper.configuration.properties.UserDefinedInstanceProperty.JARS_BUCKET;
+import static sleeper.configuration.properties.UserDefinedInstanceProperty.REGION;
+import static sleeper.configuration.properties.UserDefinedInstanceProperty.SUBNET;
+import static sleeper.configuration.properties.UserDefinedInstanceProperty.VPC_ID;
+import static sleeper.configuration.properties.table.TableProperty.TABLE_NAME;
+import static sleeper.core.schema.SchemaTestHelper.schemaWithKey;
+
+public class UtilsTestHelper {
+    private UtilsTestHelper() {
+    }
+
+    public static Function<String, String> cdkUpgradeContextWithPropertiesFile(Path tempDir) {
+        return cdkContextWithPropertiesFile(tempDir, false);
+    }
+
+    public static Function<String, String> cdkUpdateContextWithPropertiesFile(Path tempDir) {
+        return cdkContextWithPropertiesFile(tempDir, true);
+    }
+
+    private static Function<String, String> cdkContextWithPropertiesFile(Path tempDir, boolean update) {
+        return Map.of("propertiesfile", tempDir.resolve("instance.properties").toString(),
+                "update", "" + update)::get;
+    }
```

Comment:
This isn't quite valid. The usual way the CDK will be called is with just the properties file and no flags. If the system is being upgraded from one version to another, a flag should be set, either called upgrade or update. I think those should mean the same thing in this context.